content/articles/using-logistic-regression-in-pytorch-to-identify-handwritten-digits/index.md

```diff
@@ -0,0 +1,169 @@
+
+### Prerequisites
+1. Install [PyTorch](https://pytorch.org/) into your Python environment.
+2. Having some knowledge of the Python programming language is a good idea.
+
+### Table of contents
+- [Library functions and objects](#library-functions-and-objects)
+- [Defining our hyperparameters](#defining-our-hyperparameters)
+- [Building The Neural Network](#building-the-neural-network)
+- [References ](#references )
+- [Conclusion](#conclusion)
+ 
+### Library functions and objects
+Importing library functions and objects into your code is necessary once your installation is complete.
+Here are the functions and objects to import:
+
+- The **torch.nn** module: Contains code thatwe will be using in our model.
+- The **torchvision.datasets**: Includes MNIST dataset of handwritten digits that we shall be using here.
+- The **torchvision.transforms**: We shall be using it to transform from images to PyTorch tensors.
+- The **torch.autograd**: This will be used to define our tensors.
+
+Type the code below to import the library functions and objects;
+```Python
+import torch
+import torch.nn as tn
+import torch.nn.functional as tF
+import torch.optim as optim
+import torchvision.datasets as dsets
+import torchvision.transforms as transforms
+from torch.autograd import Variable
+
+```
+After importing we shall now download and load our dataset to memory. This will be done in the following code:
+```Python
+# MNIST Dataset 
+train_dataset = dsets.MNIST(root ='./data',train = True, download = True)
+
+test_dataset = dsets.MNIST(root ='./data', train = False, transform = transforms.ToTensor())
+
+# Dataset Loader 
+train_loader = torch.utils.data.DataLoader(dataset = train_dataset, shuffle = True)
+
+test_loader = torch.utils.data.DataLoader(dataset = test_dataset,shuffle = False)
+```
+### Defining our hyperparameters
+Let us define our hyperparameters:
+```Python
+input_size = 784
+num_classes = 10
+num_epochs = 5
+batch_size = 100
+learning_rate = 0.001
+
+```
+The image size that we will use for our image will be 28*28. This means that our input size will be 784. In addition, there are ten digits in this. Thus we can generate ten alternative results. In this way, `number_classes` will be set to ten. In addition, we'll run the full dataset through five iterations of training. Finally, we will train in small batches of 100 images each so that the software does not crash because of memory overflow.
+
+After that, we'll lay out our model in the following manner. In this section, we'll define the forward pass after setting up our model as a subclass of __torch.tn.Module__. As we are building the code, we will not need to mention the softmax in the __forward()__ function because it will be determined internally during each forward run.

```

Comment:
torch.nn.Module

Reply:
Isn't it usually imported as nn? Correct me if I am wrong here. If you want to proceed with tn, explicitly mention what tn is.